ql/math/expm1.cpp

```diff
@@ -0,0 +1,54 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2023 Klaus Spanderen
+
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+#include <ql/math/expm1.hpp>
+#include <ql/math/functional.hpp>
+
+#include <cmath>
+
+namespace QuantLib {
+    std::complex<Real> expm1(const std::complex<Real>& z) {
```

Comment:
This implementation uses lots of trigonometric functions. There are implementations making only use of evaluating simple polynomials, e.g. [scipy-link](https://github.com/scipy/scipy/blob/4d50bca5eeabe650a8dc9729f32a886a48f13ba9/scipy/special/cephes/unity.c#L106).

Reply:
These implementations are for real arguments. Without further proof I'd be rather cautious to use the Pde-approximations in the complex plane.

Reply:
You are correct. 
I found this [stackoverflow answer](https://stackoverflow.com/a/42501935/10774817) giving the same formula. Regard this thread as void if you want.